tests/installation/upgrade_select.pm

```diff
@@ -42,7 +42,19 @@ sub run {
             send_key $cmd{ok};
         }
         else {
-            assert_screen 'migration_target_' . lc(get_var('SLE_PRODUCT', 'sles')), 120;
+            # Confirm default migration target matches correct base product
+            my $migration_target_base = 'migration_target_' . lc(get_var('SLE_PRODUCT', 'sles')) . lc(get_var('VERSION'));
+            assert_screen $migration_target_base, 120;
+            # Confirm other migration targets match the same base product
+            # Assume no more than 6 possible migration targets
+            send_key 'alt-p';
+            for (1 .. 5) {
+                send_key 'down';
+                assert_screen $migration_target_base;
+            }
```

Comment:
why not use `send_key_until_needlematch`?

Reply:
We need check every possible migration target could match the base product needle. According to fate#323163: It can happen that it provides two possible targets (e.g. two versions of SES compatible with the same base product).

Reply:
And I'm thinking to workaround bsc#1071185 and make migration proceed further.

Reply:
got your point, thanks